src/pickup.cpp

```diff
@@ -417,82 +366,26 @@ void Pickup::autopickup( const tripoint &p )
             autopickup( apos );
         }
     }
-
     // Bail out if this square cannot be auto-picked-up
-    if( g->check_zone( zone_type_NO_AUTO_PICKUP, p ) ) {
-        return;
-    }
-    if( local.has_flag( ter_furn_flag::TFLAG_SEALED, p ) ) {
+    if( g->check_zone( zone_type_NO_AUTO_PICKUP, p ) ||
+        local.has_flag( ter_furn_flag::TFLAG_SEALED, p ) ) {
         return;
     }
-
-    std::vector<std::list<item_stack::iterator>> stacked_here;
-    for( const item_stack::iterator &it : here ) {
-        bool found_stack = false;
-        for( std::list<item_stack::iterator> &stack : stacked_here ) {
-            if( stack.front()->display_stacked_with( *it ) ) {
-                stack.push_back( it );
-                found_stack = true;
-                break;
-            }
-        }
-        if( !found_stack ) {
-            stacked_here.emplace_back( std::list<item_stack::iterator>( { it } ) );
-        }
-    }
-
-    // Items are stored unordered in colonies on the map, so sort them for a nice display.
-    std::sort( stacked_here.begin(), stacked_here.end(), []( const auto & lhs, const auto & rhs ) {
-        return *lhs.front() < *rhs.front();
-    } );
-
-    std::vector<pickup_count> getitem( stacked_here.size() );
-
-    if( !select_autopickup_items( stacked_here, getitem ) ) {
-        // If we didn't find anything, bail out now.
+    std::vector<int> quantities;
+    std::vector<item_location> target_items = auto_pickup::select_items( here, p );
```

Comment:
Since I rarely use the autopickup feature I'm not sure about this, but with the coming introduction of a max quantity for autopickup, I guess the quantity would have to be part of the selection already? In that case you could refactor this to return `drop_locations`, which is an alias for `std::list<std::pair<item_location, int>>` used by multiselector menus.

If I'm wrong about how this is going to work, you can disregard this. It's also not necessary for this PR, more a "while you're at it" thing. It might as well go into #54894 if it's necessary.

(As an aside, this might also be the last place necessary to refactor `pickup_activity_actor` to use `drop_locations` instead of two separate vectors. And yes, `drop_locations` is a wildly inaccurate name at this point.)

Reply:
I've refactored the code as you requested in 8aeb408590782c38f84c3e0ec9bba7656ac03359